pkg/network/node/networkpolicy.go

```diff
@@ -276,15 +282,30 @@ func (np *networkPolicyPlugin) DeleteNetNamespace(netns *osdnv1.NetNamespace) {
 	np.updateMatchCache(npns)
 }
 
-func (np *networkPolicyPlugin) SetUpPod(podIP string) error {
+func (np *networkPolicyPlugin) SetUpPod(pod *corev1.Pod, podIP string) error {
+	np.lock.Lock()
+	defer np.lock.Unlock()
+	np.localPodIPs[pod.UID] = podIP
 	otx := np.node.oc.NewTransaction()
 	// when a pod is created isolate the pod until the current network policy rules can be
 	// evaluated
-	otx.AddFlow("table=27, priority=500, cookie=1, ip, nw_src=%s, actions=drop", podIP)
-	otx.AddFlow("table=80, priority=500, cookie=1, ip, nw_src=%s, actions=drop", podIP)
+	otx.AddFlow("table=27, priority=500, cookie=%d, ip, nw_src=%s, actions=drop", NetPolIsolationCookie, podIP)
+	otx.AddFlow("table=80, priority=500, cookie=%d, ip, nw_src=%s, actions=drop", NetPolIsolationCookie, podIP)
 	if err := otx.Commit(); err != nil {
-		return fmt.Errorf("Error syncing OVS flows to isolate pods %v", err)
+		return fmt.Errorf("error syncing OVS flows to isolate pods %v", err)
+	}
+
+	syncTriggered := np.refreshPodNetworkPolicies(pod)
+	// if no network policies update needed for this pod, remove drop rules
+	if !syncTriggered {
+		otx := np.node.oc.NewTransaction()
+		otx.DeleteFlows("table=27, cookie=%d/-1, ip, nw_src=%s", NetPolIsolationCookie, podIP)
+		otx.DeleteFlows("table=80, cookie=%d/-1, ip, nw_src=%s", NetPolIsolationCookie, podIP)
```

Comment:
as mentioned on slack, you should be able to just not add the flows in the first place if a sync isn't triggered